packages/core/src/boot/init.ts

```diff
@@ -48,6 +49,11 @@ export interface BuildEnv {
 }
 
 export function commonInit(initConfiguration: InitConfiguration, buildEnv: BuildEnv) {
+  const enableExperimentalFeatures = Array.isArray(initConfiguration.enableExperimentalFeatures)
+    ? initConfiguration.enableExperimentalFeatures
+    : []
+  setEnabledExperimentalFeatures(enableExperimentalFeatures)
+
   const configuration = buildConfiguration(initConfiguration, buildEnv)
   const internalMonitoring = startInternalMonitoring(configuration)
```

Comment:
it could be nice to extract experimental features init to its own function to keep the same level of abstraction inside commonInit